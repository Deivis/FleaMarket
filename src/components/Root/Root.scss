@import '../../styles/_variables';

.root__version {
  position: fixed;
  left: 5px;
  bottom: 5px;
  font-size: 0.8em;
}
