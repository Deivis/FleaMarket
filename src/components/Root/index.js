import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

const Root = ({ store, history, routes }) => (
  <div>
    <Provider store={store}>
      <BrowserRouter
        basename={process.env.APP_NAME}
        history={history}
      >
        {routes(store)}
      </BrowserRouter>
    </Provider>
    <span className="root__version">Version {process.env.npm_package_version}</span>
  </div>
);

Root.propTypes = {
  store: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({}).isRequired,
  routes: PropTypes.func.isRequired,
};

export default Root;
