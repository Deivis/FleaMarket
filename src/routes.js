import { Route, Switch } from 'react-router-dom';
import React from 'react';

import NotFound from './components/NotFound';
import Home from './modules/home';
import Item from './modules/item';
import Cart from './modules/cart';
import Checkout from './modules/checkout';
import Payment from './modules/payment';
import Summary from './modules/summary';

const routes = () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route exact path="/item/:id" component={Item} />
    <Route exact path="/cart" component={Cart} />
    <Route exact path="/checkout/:id" component={Checkout} />
    <Route exact path="/checkout/:id/payment" component={Payment} />
    <Route exact path="/checkout/:id/summary" component={Summary} />
    <Route component={NotFound} />
  </Switch>
);

export default routes;
