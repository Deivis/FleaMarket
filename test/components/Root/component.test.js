import React from 'react';
import renderer from 'react-test-renderer';
import Root from '../../../src/components/Root';

import configureStore from '../../../src/utils/configStore';
import routes from '../../../src/routes';

const store = configureStore();
const history = store.browserHistory;
const rootProps = { store, history, routes };

describe('Root Tests', () => {
  test('Root Render', () => {
    const tree = renderer
      .create(<Root {...rootProps} routes={routes} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
