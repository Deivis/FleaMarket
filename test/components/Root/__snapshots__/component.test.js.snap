// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Tests Root Render 1`] = `
<div>
  <div
    className="overlay__not-found"
  >
    <h1>
       Page not found 
    </h1>
  </div>
  <span
    className="root__version"
  >
    Version 
    1.0.1
  </span>
</div>
`;
